web/packages/design/src/DataTable/StyledTable.tsx

```diff
@@ -89,6 +93,23 @@ export const StyledTable = styled.table(
   borderRadius
 );
 
+// When `border-collapse: collapse` is set on a table element, Safari incorrectly renders row border with alpha channel.
+// It looks like the collapsed border was rendered twice, that is, opacity 0.07 looks like opacity 0.14 (this is more visible
+// on dark theme).
+// Sometimes, there is also an artifact visible after hovering the rows - some of them have correct border color, some not.
+//
+// `getSolidRowBorderColor` is a workaround. The colors below were created by lightening or darkening the table background by
+// the value of theme.colors.spotBackground[0]
+function getSolidRowBorderColor(theme) {
+  const alpha = decomposeColor(theme.colors.spotBackground[0]).values[3];
+  if (theme.name === 'dark') {
+    return lighten(theme.colors.levels.surface, alpha);
+  }
+  if (theme.name === 'light') {
+    return darken(theme.colors.levels.surface, alpha);
+  }
+}
```

Comment:
Any opinions on hardcoding the final values here? I'm a little worried that it may cause some perf issues when rendering a lot of rows.

Reply:
What do you mean by hardcoding them here?

I actually don't know how styled components work with these custom functions using props, are they going to compile the style only once and then extract it to a CSS file?

Reply:
>What do you mean by hardcoding them here?

return "rgb(49,58,100)"; 🙈

But nvm, it doesn't make any sense, because:
1. We create style for an entire table, not for each row separately.
2. The style is probably re-complied only when the prop changes, I added `console.log` to `getSolidRowBorderColor` and it was only invoked again only when I changed the theme.

Btw, styled-components doesn't create a CSS file, but:
>styled-components generates an actual stylesheet with classes, and attaches those classes to the DOM nodes of styled components via the className prop. It injects the generated stylesheet at the end of the head of the document during runtime.